zrml/prediction-markets/src/tests.rs

```diff
@@ -871,6 +948,100 @@ fn full_scalar_market_lifecycle() {
+
+        // Check balances before redeeming (just to make sure that our tests are based on correct
+        // assumptions)!
+        assert_eq!(Balances::free_balance(&CHARLIE), 900 * BASE);
+        assert_eq!(Balances::free_balance(&EVE), 1000 * BASE);
+
+        assert_ok!(PredictionMarkets::redeem_shares(Origin::signed(CHARLIE), 0));
+        assert_ok!(PredictionMarkets::redeem_shares(Origin::signed(EVE), 0));
+        let assets = PredictionMarkets::outcome_assets(0, &MarketCommons::market(&0).unwrap());
+        for asset in assets.iter() {
+            assert_eq!(Tokens::free_balance(*asset, &CHARLIE), 0);
+            assert_eq!(Tokens::free_balance(*asset, &EVE), 0);
+        }
+        assert_eq!(Balances::free_balance(&CHARLIE), 900 * BASE);
+        assert_eq!(Balances::free_balance(&EVE), 1100 * BASE);
+    })
+}
+
+#[test]
+fn scalar_market_correctly_resolves_on_out_of_range_outcomes_above_threshold() {
+    ExtBuilder::default().build().execute_with(|| {
+        simple_create_scalar_market::<Runtime>(
+            MarketCreation::Permissionless,
+            0..100,
+            ScoringRule::CPMM,
+        );
+        assert_ok!(PredictionMarkets::buy_complete_set(Origin::signed(CHARLIE), 0, 100 * BASE));
+        assert_ok!(Tokens::transfer(
+            Origin::signed(CHARLIE),
+            EVE,
+            Asset::ScalarOutcome(0, ScalarPosition::Short),
+            100 * BASE
+        ));
+        // (Eve now has 100 SHORT, Charlie has 100 LONG)
+
+        run_to_block(100);
+        assert_ok!(PredictionMarkets::report(Origin::signed(BOB), 0, OutcomeReport::Scalar(250)));
+        let market_after_report = MarketCommons::market(&0).unwrap();
+        assert_eq!(market_after_report.report.is_some(), true);
+        let report = market_after_report.report.unwrap();
+        assert_eq!(report.at, 100);
+        assert_eq!(report.by, BOB);
+        assert_eq!(report.outcome, OutcomeReport::Scalar(250));
+
+        run_to_block(150);
+        let market_after_resolve = MarketCommons::market(&0).unwrap();
+        assert_eq!(market_after_resolve.status, MarketStatus::Resolved);
+
+        // Check balances before redeeming (just to make sure that our tests are based on correct
+        // assumptions)!
+        assert_eq!(Balances::free_balance(&CHARLIE), 900 * BASE);
+        assert_eq!(Balances::free_balance(&EVE), 1000 * BASE);
+
+        assert_ok!(PredictionMarkets::redeem_shares(Origin::signed(CHARLIE), 0));
+        assert_ok!(PredictionMarkets::redeem_shares(Origin::signed(EVE), 0));
+        let assets = PredictionMarkets::outcome_assets(0, &MarketCommons::market(&0).unwrap());
+        for asset in assets.iter() {
+            assert_eq!(Tokens::free_balance(*asset, &CHARLIE), 0);
+            assert_eq!(Tokens::free_balance(*asset, &EVE), 0);
+        }
+        assert_eq!(Balances::free_balance(&CHARLIE), 1000 * BASE);
+        assert_eq!(Balances::free_balance(&EVE), 1000 * BASE);
+    })
+}
+
```

Comment:
Nothing wrong here, just a lot of common code that could be put into a common function

Reply:
:thumbsup: Yeah, resolved!